### PR TITLE
No End Products on non-comp SC and HLR

### PR DESCRIPTION
**Pull request by pkarman**
connect #7985 

### Description
Adds new method `DecisionReview.non_comp?` and applies it to UI text and processing logic.

Note that `Appeal` now has a hardcoded `benefit_type` in order to satisfy the `non_comp?` method. That is intended, and will be resolved when #5882 is implemented.

### CODE REVIEW

File: client/app/intake/pages/higherLevelReview/finish.jsx [hunk 1]
```diff
@@ -19,14 +19,22 @@ class FinishNextButton extends React.PureComponent {
     );
   }
 
+  buttonText = () => {
+    if (this.props.higherLevelReview.nonComp) {
+      return `Establish ${FORM_TYPES.HIGHER_LEVEL_REVIEW.shortName}`;
+    }
+
+    return 'Establish EP';
```
Comment (jiahuang) [inline client/app/intake/pages/higherLevelReview/finish.jsx]: could shorten this to
return `Establish ${this.props.higherLevelReview.nonComp ? FORM_TYPES.HIGHER_LEVEL_REVIEW.shortName : "EP"};`

File: spec/feature/intake/supplemental_claim_spec.rb [hunk 2]
```diff
@@ -780,13 +780,33 @@ def check_row(label, text)
...
+      sc, = start_supplemental_claim(veteran, is_comp: false)
       visit "/intake/add_issues"
 
       expect(page).to have_content("Add / Remove Issues")
       check_row("Form", Constants.INTAKE_FORM_NAMES.supplemental_claim)
       check_row("Benefit type", "Education")
+
+      expect(page).to_not have_content("Left knee granted")
+
+      click_intake_add_issue
```
Comment (jiahuang) [inline spec/feature/intake/supplemental_claim_spec.rb]: maybe there should also be a test that sets up decision issues? those decision issues should show up on the add rating issues modal right?

File: client/app/intake/pages/higherLevelReview/finish.jsx (hunk 1 above)
Comment (pkarman) [reply client/app/intake/pages/higherLevelReview/finish.jsx]: true. I find a single-line ternary expression like that harder to read, personally.

File: spec/feature/intake/supplemental_claim_spec.rb (hunk 2 above)
Comment (pkarman) [reply spec/feature/intake/supplemental_claim_spec.rb]: do you mean as part of this `scenario`? or to cover some other aspect of this PR?

File: client/app/intake/pages/higherLevelReview/finish.jsx (hunk 1 above)
Comment (jiahuang) [reply client/app/intake/pages/higherLevelReview/finish.jsx]: i'm like a 2/10 on caring about this so if its easier to not be ternary that's fine with me.

File: spec/feature/intake/supplemental_claim_spec.rb (hunk 2 above)
Comment (jiahuang) [reply spec/feature/intake/supplemental_claim_spec.rb]: can either be part of this scenario or creating a new one. seems easiest just to add it in here
